### ISSUE TITLE
When user is not already logged into GitHub, they are not redirected to callback URL

### ISSUE THREAD

**Issue by chrisvfritz** [intridea/omniauth-github#56]
I'm not sure what the source of the issue is. Everything works fine when a user is already logged in to GitHub, but if they are prompted to log in, the redirect is to github.com instead. In case the login parameters might help diagnose the problem, here they are:

```
?return_to=%2Flogin%2Foauth%2Fauthorize%3Fclient_id%3D81132bde02e11f7f2e67%26redirect_uri%3Dhttp%253A%252F%252Flansingcodelab.com%252Fauth%252Fgithub%252Fcallback%26response_type%3Dcode%26scope%3Duser%253Aemail%252Cnotifications%252Cpublic_repo%26state%3Dde3ea4e1be77b433b392ce879ae075f85dfd25d5cacd37cc
```

**Comment by chrisledet** [comment]
I ran into this problem on while serving requests via https. I noticed callback url generated by omniauth was using http although I was using https. I found the problem was due to my minimal nginx configuration. I'm using nginx to proxy pass requests to a rails server. I was not passing the [proper headers](https://github.com/intridea/omniauth/blob/32c4165919ab34189faa443cea66dfb11ba4fbff/lib/omniauth/strategy.rb#L500-L504) for the site to detect if it was under https. Once I added those, this problem went away.
